_posts/2019-03-01-resizing-persistent-volumes.md

```diff
@@ -0,0 +1,267 @@
+/dev/xvda2      120G  4.7G  116G   4% /etc/hosts
+tmpfs           3.9G  4.0K  3.9G   1% /opt/kafka/cluster-ca-certs
+tmpfs           3.9G   24K  3.9G   1% /opt/kafka/broker-certs
+tmpfs           3.9G  4.0K  3.9G   1% /opt/kafka/client-ca-certs
+/dev/xvdbe       99G   61M   99G   1% /var/lib/kafka/data
+tmpfs           3.9G   12K  3.9G   1% /run/secrets/kubernetes.io/serviceaccount
+tmpfs           3.9G     0  3.9G   0% /proc/acpi
+tmpfs           3.9G     0  3.9G   0% /proc/scsi
+tmpfs           3.9G     0  3.9G   0% /sys/firmware
+```
+
+Notice the volume `/dev/xvdbe` which is the persistent volume with the Kafka message logs and has total capacity of 99G.
+As you can see I have still a lot of free disk space.
+But I want to increase the disk size anyway.
+Let's say to 200Gi.
+
+As the first step, I have to go to the Persistent Volume Claims and edit the requested capacity.
+In my case I will change the requested capacity from 100Gi to 200Gi.
+You can apply this change to all PVC of your Kafka cluster.
+
+```yaml
+# ...
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 200Gi
+  volumeName: pvc-68b4be67-3a8a-11e9-bf2b-067919d3dbb8
+  storageClassName: gp2
+  dataSource: null
+# ...
+```
+
+After you apply these changes, the `status` field of the PVC should change to something like this:
+
+```yaml
+#...
+status:
+  phase: Bound
+  accessModes:
+    - ReadWriteOnce
+  capacity:
+    storage: 100Gi
+  conditions:
+    - type: Resizing
+      status: 'True'
+      lastProbeTime: null
+      lastTransitionTime: '2019-02-27T12:25:15Z'
+```
+
+As you can see, the capacity is still 100Gi, but Kubernetes / OpenShift are now resizing the volume.
+Once the resizing of the volume is finished, the `status` will change to this:
+
+```yaml
+# ...
+status:
+  phase: Bound
+  accessModes:
+    - ReadWriteOnce
+  capacity:
+    storage: 100Gi
+  conditions:
+    - type: FileSystemResizePending
+      status: 'True'
+      lastProbeTime: null
+      lastTransitionTime: '2019-02-27T12:25:30Z'
+      message: >-
+        Waiting for user to (re-)start a pod to finish file system resize of
+        volume on node.
+```
+
+As you can read in the message, the volume has been resized.
+But the file system is still only 100Gi big and needs to be resized as well.
+The resizing of the file system cannot be done while the pod is running - it will be done automatically when the pod is restarted. 
+
+The easiest way how to restart the Kafka pods to resize the file system is to tell Strimzi to perform a rolling update of the Kafka Statefulset.
+You can do that using the [`strimzi.io/manual-rolling-update=true`annotation](https://strimzi.io/docs/latest/full.html#proc-manual-rolling-update-kafka-deployment-configuration-kafka).
+
+```console
+kubectl annotate statefulset cluster-name-kafka strimzi.io/manual-rolling-update=true
+```
+
+After you set this annotation, Strimzi will roll all the Kafka pods one by one.
```

Comment:
Mention that it might take upto a reconciliation interval for this to happen because the CO is not watching the SS?